### PR TITLE
Potential bug fix

### PR DESCRIPTION
**Pull request by ingemarberg**
Hi @SethClydesdale,

I have spotted a weird potential bug where spelling practice and fill-in-the-blanks exercises containing '。' do not seem to be generated properly. This occurs in for example lesson-8/vocab-7 (3rd ed.). It seems as every '。' is filtered out in the spelling practice exercises. The answers for the fill-in-the-blank exercises are, on the other hand, generated twice, once with '。' and once without. I guess this might be intentional in order to allow for more flexibility in the answers. If this is the case, it would perhaps be better if the answer containing furigana were displayed instead of the one with punctuation.

This pull request contains a fix for this potential bug. However, please do note that I am not really a programmer and that this might break something else. I have tried checking against other exercises and the fix seems to work properly. Please let me know if I have made some mistakes. Do not hesitate to close this pull request if I have misunderstood and things are already working as intended.

As always, thanks for this wonderful resource!

### CODE REVIEW

**Comment by SethClydesdale** [comment]
Good evening @ingemarberg,

Thanks for bringing this to my attention! You're correct that these regular expressions were setup to filter out certain characters in an attempt to improve flexibility, at least for the fill-type quizzes. As for the spelling practice ones, I think preserving the punctuation _would_ be more ideal.

Anyway, I do agree that the furigana should be visible as the second answer. The answer structure is handled a little further down around [here](https://github.com/SethClydesdale/genki-study-resources/blob/cab952e87fee609011cf4291f73228c55b579d76/resources/javascript/genki.js#L380). Changing it to the following _should_ probably fix that by showing the two filtered answers first (kanji --> furigana), with the unfiltered ones being presented as additional hidden answers.
```javascript
                      def[0].replace(filter, '')+ // answer 1
                      (def[1] ? '|' + def[1].replace(filter, '')+ // answer 2 (usu. furigana)
                      (filter.test(def[0]) ? '|' + def[0] : '')+ // unfiltered answer 1
                      (filter.test(def[1]) ? '|' + def[1] : '') : '')+ // unfiltered answer 2
```

Haven't tested it, yet, but might take a closer look at it sometime tomorrow to see how it fairs, since it's getting pretty late.

**Comment by SethClydesdale** [comment]
Hey, sorry for the late reply; I was on vacation.

The issue with furigana not being visible as the second answer _should_ be resolved via https://github.com/SethClydesdale/genki-study-resources/commit/5fb6d67a63495b5cd92a09676130e279a1bfabb7. This will also preserve the periods on spelling practice exercises.

The initial solution I posted above would have worked, however, [there's a bug with cached regexes when used with the test() function](http://blog.stevenlevithan.com/archives/es3-regexes-broken) which cut off the 4th answer and drove me mad for at least an hour before I decided to ask my good friend Google what unholy sorcery was going on. So the cached filter had to be scrapped in favor of initializing a new one for each test/replace instance.

TL;DR
Everything SHOULD be working properly now, hopefully haha.

Thanks again for bringing this to my attention. If you notice any bugs let me know.

**Comment by ingemarberg** [comment]
Hi @SethClydesdale,

It's great that your were able to fix this bug. Thanks for all your hard work! I'll let you know if I spot any bugs.
